docs/DesignDocs/CustomAttestation.md

```diff
@@ -0,0 +1,849 @@
+     * @param[in] custom_claims_size The size in bytes of custom_claims.
+     * @param[out] evidence_buffer An output pointer that will be assigned the
+     * address of the evidence buffer.
+     * @param[out] evidence_buffer_size A pointer that points to the size of the
+     * evidence buffer.
+     * @retval OE_OK on success.
+     */
+    oe_result_t (*get_evidence)(
+        oe_attestaton_plugin_t* plugin_context,
+        const uint8_t* custom_claims,
+        size_t custom_claims_size,
+        uint8_t** evidence_buffer,
+        size_t* evidence_buffer_size);
+
+    /**
+     * Frees the generated attestation evidence.
+     *
+     * @param[in] plugin_context A pointer to the attestation plugin struct.
+     * @param[in] evidence_buffer A pointer to the evidence buffer.
+     * @retval OE_OK on success.
+     */
+    oe_result_t (*free_evidence)(
+        oe_attestaton_plugin_t* plugin_context,
+        uint8_t* evidence_buffer);
+
+    /**
+     * Generates the attestation collateral, which is defined as the data
+     * produced outside of the enclave that is used part of the verification
+     * process.
+     *
+     * @param[in] plugin_context A pointer to the attestation plugin struct.
+     * @param[out] collateral_buffer An output pointer that will be assigned the
+     * address of the collateral buffer.
+     * @param[out] collateral_buffer_size A pointer that points to the size of the
+     * collateral buffer.
+     * @retval OE_OK on success.
+     */
+    oe_result_t (*get_collaterals)(
+        oe_attestaton_plugin_t* plugin_context,
+        uint8_t** collateral_buffer,
+        size_t* collateral_buffer_size);
+
+    /**
+     * Frees the generated attestation collateral.
+     *
+     * @param[in] plugin_context A pointer to the attestation plugin struct.
+     * @param[in] collateral_buffer A pointer to the collateral buffer.
+     * @retval OE_OK on success.
+     */
+    oe_result_t (*free_collaterals)(
+        oe_attestaton_plugin_t* plugin_context,
+        uint8_t* collateral_buffer);
+
+    /**
+     * Verifies the attestation evidence and returns the claims contained in
+     * the evidence.
+     *
+     * @param[in] plugin_context A pointer to the attestation plugin struct.
+     * @param[in] evidence_buffer The evidence buffer.
+     * @param[in] evidence_buffer_size The size of evidence_buffer in bytes.
+     * @param[in] collateral_buffer The collateral buffer.
+     * @param[in] collateral_buffer_size The size of collateral_buffer in bytes.
+     * @param[out] claims The list of claims.
+     * @param[out] claims_size The size of claims.
+     * @retval OE_OK on success.
+     */
+    oe_result_t (*verify_evidence)(
+        oe_attestaton_plugin_t* plugin_context,
+        const uint8_t* evidence_buffer,
+        size_t evidence_buffer_size,
+        const uint8_t* collateral_buffer,
+        size_t collateral_buffer_size,
+        uint_t** claims,
+        size_t* claims_size);
+};
```

Comment:
I think we will probably also need APIs somewhere to verify attestation results (i.e., responses from a remote verifier like AAS).   Would those belong in the OE SDK or in something on top of the OE SDK?  (I don't know)